apps/api/src/app/shared/services/query-parser/query-validator.service.ts

```diff
@@ -0,0 +1,245 @@
+import { AdditionalOperation, RulesLogic } from 'json-logic-js';
+
+import { COMPARISON_OPERATORS, JsonComparisonOperatorEnum, JsonLogicOperatorEnum } from './types';
+
+type QueryIssue = {
+  message: string;
+  path: number[];
+  type: QueryIssueTypeEnum;
+};
+
+export enum QueryIssueTypeEnum {
+  INVALID_STRUCTURE = 'INVALID_STRUCTURE',
+  MISSING_VALUE = 'MISSING_VALUE',
+}
+
+export class QueryValidatorService {
```

Comment:
The service validates the structure of the step conditions and returns a list of issues, if any.
The service itself is not connected to workflow or step, so we can potentially reuse it in other places in the future.

Reply:
The unit tests file is above.